resource-managers/yarn/src/main/scala/org/apache/spark/deploy/yarn/Client.scala

```diff
@@ -811,10 +811,18 @@ private[spark] class Client(
 
     // Finally, update the Spark config to propagate PYTHONPATH to the AM and executors.
     if (pythonPath.nonEmpty) {
-      val pythonPathStr = (sys.env.get("PYTHONPATH") ++ pythonPath)
+      val pythonPathStr = (sys.env.get("PYTHONPATH") ++=: pythonPath)
         .mkString(ApplicationConstants.CLASS_PATH_SEPARATOR)
-      env("PYTHONPATH") = pythonPathStr
-      sparkConf.setExecutorEnv("PYTHONPATH", pythonPathStr)
+      val newValue =
```

Comment:
This seems unnecessary. What you're trying to do here is concatenate the values of the current `PYTHONPATH` env variable, the `PYTHONPATH` created by Spark, and the `PYTHONPATH` set in `spark.yarn.appMasterEnv`.

That's, respectively, `sys.env.get("PYTHONPATH")`, `pythonPath`, and `env.get("PYTHONPATH")`. So just concatenate those.

Reply:
I would have done it, but I need the pythonPathStr variable to set it in executorEnv so have not modified that bit. Let me know if you think otherwise.